numpy/matrixlib/tests/test_matrix_linalg.py

```diff
@@ -0,0 +1,96 @@
+""" Test functions for linalg module using the matrix class."""
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+
+from numpy.linalg.tests.test_linalg import (
+    LinalgCase, apply_tag, _TestNorm2D, TestQR as _TestQR, LinalgTestCase,
+    SolveCases, InvCases, EigvalsCases, EigCases, SVDCases, CondCases,
+    PinvCases, DetCases, LstsqCases)
+
+
+CASES = []
+
+# square test cases
+CASES += apply_tag('square', [
+    LinalgCase("0x0_matrix",
+               np.empty((0, 0), dtype=np.double).view(np.matrix),
+               np.empty((0, 1), dtype=np.double).view(np.matrix),
+               tags={'size-0'}),
+    LinalgCase("matrix_b_only",
+               np.array([[1., 2.], [3., 4.]]),
+               np.matrix([2., 1.]).T),
+    LinalgCase("matrix_a_and_b",
+               np.matrix([[1., 2.], [3., 4.]]),
+               np.matrix([2., 1.]).T),
+])
+
+# hermitian test-cases
+CASES += apply_tag('hermitian', [
+    LinalgCase("hmatrix_a_and_b",
+               np.matrix([[1., 2.], [2., 1.]]),
+               None),
+])
+# No need to make generalized or strided cases for matrices.
+
+
+class MatrixTestCase(LinalgTestCase):
+    TEST_CASES = CASES
+
+
+class TestSolve(SolveCases, MatrixTestCase):
+    pass
+
+
+class TestInv(InvCases, MatrixTestCase):
+    pass
+
+
+class TestEigvals(EigvalsCases, MatrixTestCase):
+    pass
+
+
+class TestEig(EigCases, MatrixTestCase):
+    pass
+
+
+class TestSVD(SVDCases, MatrixTestCase):
+    pass
+
+
+class TestCond(CondCases, MatrixTestCase):
+    pass
+
+
+class TestPinv(PinvCases, MatrixTestCase):
+    pass
```

Comment:
I'd be temped to just combine these onto one line, and remove all the spacing between them. Perhaps PEP8 disagrees though

Reply:
It does...

Reply:
That's a shame. Suppose you could get around that with `TestPinv = type('TestPinv', (PinvCases, MatrixTestCase), {})` to force it onto one line, but that's super ugly